### PR TITLE
AC_PrecLand: ignore msg in particular frame indicating report only

### LINKED ISSUE #3348: Copter: Add precision landing status to telemetry
ArduCopter currently logs precision landing to DataFlash log. But it will be very useful to also see precision landing status in real time telemetry through Mavlink.

### PR DESCRIPTION
**Pull request by chobitsfan**
fix #3348

continue work of #9125 by @fnoop 
This PR try to solve issue mentioned by @peterbarker 

> If you have two vehicles which can see each other's traffic you may start
to get cross-pollination between the two.  There's no target fields in
this message.

It use frame field to indicate this message is reporting back to GCS. Similar to the usage of MAV_FRAME_MISSION
> MAV_FRAME_MISSION:	NOT a coordinate frame, indicates a mission command.

### CODE REVIEW

File: ArduCopter/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -454,6 +463,7 @@ static const ap_message STREAM_EXTRA3_msgs[] = {
     MSG_VIBRATION,
     MSG_RPM,
     MSG_ESC_TELEMETRY,
+    MSG_LANDING_TARGET,
```
Comment (khancyr) [inline ArduCopter/GCS_Mavlink.cpp]: I wouldn't add it to a stream. We now support requesting a single message !

Comment (chobitsfan) [reply ArduCopter/GCS_Mavlink.cpp]: OK

**Comment by fnoop** [comment]
@chobitsfan I don't think using the 'frame' field to mark reporting is a good idea - it's supposed to be an enum of MAV_FRAME, and you would also lose the frame info when reading the message.  Similarly if you used 'type' field.
It's the same problem with TIMESYNC and probably other messages - it should be fixed by adding target fields to the mavlink message, and/or better a specfic 'info/reporting' field.  I think this is out of scope of the original PR, which is for reporting messages.

**Comment by fnoop** [comment]
ps. this limitation should be documented in the mavlink notes (https://github.com/mavlink/mavlink-devguide/blob/master/en/services/landing_target.md) and the respective autopilot notes for precland.  I originally suggested adding a separate message for info, but was advised we should reuse this message.

**Comment by chobitsfan** [comment]
hi @fnoop thank you
I agree with you that adding target fields is the best solution. However, it may not be easy to modify existing message. MAVLink maintainers may not accept this change because it will break compatibility. 

> and you would also lose the frame info when reading the message

Yes, but when it is used for drone reporting to GCS, GCS may probably know what frame expected.

**Comment by chobitsfan** [comment]
I think @fnoop is right. Maybe we should adding a separate message for this purpose.

**Pull request closed without merging**